niworkflows/workflows/epi/refmap.py

```diff
@@ -0,0 +1,220 @@
+    -------
+    epiref : :obj:`str`
+        Path of the generated EPI reference file.
+    xfms : :obj:`list` of :obj:`str`
+        List of rigid-body transforms in LTA format to resample from
+        the reference volume of each run into the ``epiref`` reference.
+    volumes : :obj:`list` of :obj:`str`
+        List of paths to the reference volume generated per input run.
+    gs_drift : :obj:`list` of :obj:`list` of :obj:`float`
+        A list of global signal drift factors for the set of volumes selected
+        for averaging, per run.
+    n_dummy : :obj:`list` of :obj:`int`
+        Number of nonsteady states at the beginning of each run (only BOLD with
+        ``auto_bold_nss=True``)
+
+    See Also
+    --------
+    Discussion and original flowchart at `nipreps/niworkflows#601
+    <https://github.com/nipreps/niworkflows/issues/601>`__.
+
+    """
+    from nipype.interfaces.ants import N4BiasFieldCorrection
+    from ...utils.connections import listify
+
+    from ...interfaces.bold import NonsteadyStatesDetector
+    from ...interfaces.freesurfer import StructuralReference
+    from ...interfaces.header import ValidateImage
+    from ...interfaces.images import RobustAverage
+    from ...interfaces.nibabel import IntensityClip
+
+    wf = Workflow(name=name)
+
+    inputnode = pe.Node(
+        niu.IdentityInterface(fields=["in_files", "t_masks"]), name="inputnode"
+    )
+    outputnode = pe.Node(
+        niu.IdentityInterface(
+            fields=["epiref", "xfms", "volumes", "gs_drift"]
+            + ["n_dummy"] * auto_bold_nss
+        ),
+        name="outputnode",
+    )
+
+    validate_nii = pe.MapNode(
+        ValidateImage(), name="validate_nii", iterfield=["in_file"]
+    )
+
+    run_avgs = pe.MapNode(
+        RobustAverage(), name="run_avgs", mem_gb=1, iterfield=["in_file", "t_mask"]
+    )
+
+    clip_avgs = pe.MapNode(IntensityClip(), name="clip_avgs", iterfield=["in_file"])
+
+    # de-gradient the fields ("bias/illumination artifact")
+    n4_avgs = pe.MapNode(
+        N4BiasFieldCorrection(
+            dimension=3,
+            copy_header=True,
+            n_iterations=[50] * 5,
+            convergence_threshold=1e-7,
+            shrink_factor=4,
+        ),
+        n_procs=omp_nthreads,
+        name="n4_avgs",
+        iterfield=["input_image"],
+    )
+    clipper_post = pe.MapNode(
+        IntensityClip(p_min=2.0, p_max=100.0),
+        name="clipper_post",
+        iterfield=["in_file"],
+    )
+
+    epi_merge = pe.Node(
+        StructuralReference(
+            auto_detect_sensitivity=True,
+            initial_timepoint=1,  # For deterministic behavior
+            intensity_scaling=True,  # 7-DOF (rigid + intensity)
+            subsample_threshold=200,
+            fixed_timepoint=True,
+            no_iteration=True,
+            transform_outputs=True,
+        ),
+        name="epi_merge",
+    )
+
+    post_merge = pe.Node(niu.Function(function=_post_merge), name="post_merge")
```

Comment:
This is just a conversion to NIfTI? Can't we tell `mri_robust_template` to output a NIfTI?

`post_merge` tells me nothing. If this step is necessary and no readable variable name springs to mind, a comment would be helpful.

Reply:
> This is just a conversion to NIfTI? Can't we tell mri_robust_template to output a NIfTI?

It used to be just a conversion, but I also stuck the clipping here so that both operations were made optional.

> post_merge tells me nothing. If this step is necessary and no readable variable name springs to mind, a comment would be helpful.

I'll add a comment.

The idea is that post_merge does nothing if only one input was fed into StructuralReference (and hence it just forwarded the input).

Reply:
Added comment in the function's docstring.